pkg/agent/run.go

```diff
@@ -107,6 +133,19 @@ func Run(cmd *cobra.Command, args []string) {
 		}()
 	}
 
+	_, isVenConn := preflightClient.(*client.VenConnClient)
+	if isVenConn {
+		go func() {
+			err := preflightClient.(manager.Runnable).Start(ctx)
+			if err != nil {
+				logs.Log.Fatalf("failed to start a controller-runtime component: %v", err)
+			}
+
+			// The agent must stop if the controller-runtime component stops.
+			cancel()
```

Comment:
I guess the prometheus and profiling go routines  (above) should also call cancel....but that's a bug to be fixed in a separate PR (if you agree).

Reply:
You are correct. I opened a ticket to remind us of that: https://venafi.atlassian.net/browse/VC-35567